### DISCUSSION TITLE
'ubuntu-latest' runner not found when using workflow_call between repositories

### DISCUSSION THREAD

**Discussion by PurpleBooth** [community/community#25303] [Actions]
<p>I am getting a weird error when reusing workflows</p>
<pre><code class="lang-auto">on:
  workflow_call:
jobs:
  common:
    strategy:
      matrix:
        os: [macos-latest, ubuntu-latest, windows-latest]
    runs-on: ${{ matrix.os }}
    steps:
      - echo echo hello
</code></pre>
<pre><code class="lang-auto">on:
  push:
    branches:
      - "*"
  pull_request:
  workflow_dispatch:
jobs:
  demo:
    uses: PurpleBooth/common-pipelines/.github/workflows/common.yml@main
</code></pre>
<p>Gives me this error <code>Called workflows cannot be queued onto self-hosted runners across organisations/enterprises. Failed to queue this job. Labels: 'ubuntu-latest'.</code></p>
<p>Is there a typo or are matrix run-on’s not supported for this?</p>

**Comment by ChristopherHX** [comment]
<p>I had both workflows on the same branch same repo and I was able to use matrix.os without errors in the called workflow.</p>
<aside class="onebox githubactions" data-onebox-src="https://github.com/ChristopherHX/newcomposite-sample/actions/runs/1309435631">
  <header class="source">

      <a href="https://github.com/ChristopherHX/newcomposite-sample/actions/runs/1309435631" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">
      <svg width="24" height="24" class="github-icon github-icon-success" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm16.28-2.72a.75.75 0 00-1.06-1.06l-5.97 5.97-2.47-2.47a.75.75 0 00-1.06 1.06l3 3a.75.75 0 001.06 0l6.5-6.5z"></path></svg>



  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ChristopherHX/newcomposite-sample/actions/runs/1309435631" target="_blank" rel="noopener nofollow ugc">Update main.yml</a>
    </h4>

      <div class="github-info">
        .github/workflows/main.yml <span class="github-run-number">#28</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p><em>It’s not that github wouldn’t apply changes to the actions-service at any point of time, like between our posts</em></p>

**Comment by PurpleBooth** [comment]
<p>Cool! I will give it another go tonight, this feature is going to save me a lot of time!</p>

**Comment by PurpleBooth** [comment]
<p>Still broken for me, this is from <a href="https://github.com/PurpleBooth/git-mit" rel="noopener nofollow ugc">git-mit</a> to <a href="https://github.com/PurpleBooth/common-pipelines/blob/fe83b1a0f722e116406ea72a70fef2e751902c56/.github/workflows/rust-check.yml" rel="noopener nofollow ugc">common-pipelines</a> </p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/3/6/36e58396390c5085eaaa44fb3b857d197ead9bf3.png" data-download-href="/uploads/short-url/7PDvVifVvVqkXLSJZccbpKuRV7B.png?dl=1" title="Bildschirmfoto 2021-10-06 um 07.01.45" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/133327/181090169-352e656e-f4bc-44e4-99ad-c299065d9654.png" alt="Bildschirmfoto 2021-10-06 um 07.01.45" data-base62-sha1="7PDvVifVvVqkXLSJZccbpKuRV7B" width="690" height="402" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/3/6/36e58396390c5085eaaa44fb3b857d197ead9bf3_2_690x402.png, https://aws1.discourse-cdn.com/github/optimized/3X/3/6/36e58396390c5085eaaa44fb3b857d197ead9bf3_2_1035x603.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/3/6/36e58396390c5085eaaa44fb3b857d197ead9bf3_2_1380x804.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/3/6/36e58396390c5085eaaa44fb3b857d197ead9bf3_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Bildschirmfoto 2021-10-06 um 07.01.45</span><span class="informations">2854×1666 442 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p><strong>EDIT</strong> I have just realised that this is happening on a job without strategy matrix, so idk what is going on. I get it with <a href="https://github.com/PurpleBooth/common-pipelines/blob/fe83b1a0f722e116406ea72a70fef2e751902c56/.github/workflows/docker-build.yml" rel="noopener nofollow ugc">this</a> too</p>

**Comment by ChristopherHX** [comment]
<p>It seems, the github actions service assumes if you hit the concurrency limit of 20 jobs in a repository that you are using a self-hosted runner in a called workflow.</p>
<p>Then the beta limitation: no self-hosted runner support causes any job which hit your concurrency limit to fail. I expect this will be fixed if this limitation is removed with the General Availability of this feature ( <a href="https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#limitations" class="inline-onebox" rel="noopener nofollow ugc">Reusing workflows - GitHub Docs</a> )</p>

**Comment by ljharb** [comment]
<p>Since it’s not a self-hosted runner, that seems like a bug, beta or not?</p>

**Comment by jenschelkopf** [comment, marked as answer]
<p>Hi everyone. I’m the product manager for reusable workflows.</p>
<p>Yes, this is a bug. You shouldn’t see this error message when using github-hosted runners. And I’m not sure why you are getting an error at all. I’ve filed an issue internally and we will take a look.</p>
<p>As a note, we did update the beta (and docs) last week, so self-hosted runners are now supported with reusable workflows.</p>

**Comment by ljharb** [comment]
<p>Thanks! Please let me know when I should dive back and and try again :slight_smile:</p>

**Comment by jenschelkopf** [comment]
<p>Hello - we’ve deployed a fix for this, and you shouldn’t see this error. Give it a try and let me know if it works for you.</p>

**Comment by ljharb** [comment]
<p>Awesome, that seemed to work‼ <a href="https://github.com/inspect-js/is-arguments/runs/3954168086" class="inline-onebox" rel="noopener nofollow ugc">[actions] reuse common workflows · inspect-js/is-arguments@539328a · GitHub</a></p>
<p>How “beta” should I consider this? If the current functionality I’m using there will remain the same for the duration, I’ve got 300+ repos I would dearly love to convert to using this, but I’ll hold off if there’s a chance something would intentionally change that breaks all my CI :slight_smile:</p>

**Comment by jenschelkopf** [comment]
<p>While it’s in beta, we may change parts of the syntax or API that could break your workflows.</p>

**Comment by ljharb** [comment]
<p>Fair enough :slight_smile: then I’ll avoid migrating anything for now, but I’ll explicitly say that I’m very excited to roll this out, and please come out of beta soon! :smiley:</p>

**Comment by ljharb** [comment]
<p><a href="https://github.blog/changelog/2021-11-24-github-actions-reusable-workflows-are-generally-available/" class="inline-onebox" rel="noopener nofollow ugc">GitHub Actions: Reusable workflows are generally available | GitHub Changelog</a> yay!</p>

**Comment by briantist** [comment]
<p>Suddenly seeing this today, has been working fine, no self-hosted runners in play here…</p><aside class="onebox githubactions" data-onebox-src="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1831663519">
  <header class="source">

      <a href="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1831663519" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">


      <svg width="24" height="24" class="github-icon github-icon-failure" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm8.036-4.024a.75.75 0 00-1.06 1.06L10.939 12l-2.963 2.963a.75.75 0 101.06 1.06L12 13.06l2.963 2.964a.75.75 0 001.061-1.06L13.061 12l2.963-2.964a.75.75 0 10-1.06-1.06L12 10.939 9.036 7.976z"></path></svg>

  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1831663519" target="_blank" rel="noopener nofollow ugc">Fix typo</a>
    </h4>

      <div class="github-info">
        Collection Docs <span class="github-run-number">#737</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>
<aside class="onebox githubactions" data-onebox-src="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1831663556">
  <header class="source">

      <a href="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1831663556" target="_blank" rel="noopener nofollow ugc">github.com</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Commit">


      <svg width="24" height="24" class="github-icon github-icon-failure" viewbox="0 0 24 24" aria-hidden="true"><path d="M1 12C1 5.925 5.925 1 12 1s11 4.925 11 11-4.925 11-11 11S1 18.075 1 12zm8.036-4.024a.75.75 0 00-1.06 1.06L10.939 12l-2.963 2.963a.75.75 0 101.06 1.06L12 13.06l2.963 2.964a.75.75 0 001.061-1.06L13.061 12l2.963-2.964a.75.75 0 10-1.06-1.06L12 10.939 9.036 7.976z"></path></svg>

  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/ansible-collections/community.hashi_vault/actions/runs/1831663556" target="_blank" rel="noopener nofollow ugc">Fix typo (#217)</a>
    </h4>

      <div class="github-info">
        Collection Docs <span class="github-run-number">#76</span>
      </div>
  </div>
</div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

**Comment by briantist** [comment]
<p>There seems to be something to this <a class="mention" href="/jenschelkopf">@jenschelkopf</a> , when I retried with available capacity, it worked.</p>
